### PR TITLE
HPCC-21365 Spark Cluster Configuration Options

### PR DESCRIPTION
**Pull request by g-pan**
Signed-off-by: G-Pan <greg.panagiotatos@lexisnexis.com>

@rpastrana please review
@jpmcmu please review

The compiled PDF file : http://10.240.32.243/view/Docs%207.0/job/DocDev70/54/CE=ubuntu-14.04-amd64-1/artifact/build-docs/Release/docs/EN_US/HPCC_Spark_Integration_EN_US-7.0.0-documentation0.pdf#page=7

provided for convenience to review the composed page, vs the raw XML 

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [x] My change requires a change to the documentation.
  - [x] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Testing:

Successfully completed build: http://10.240.32.243/view/Docs%207.0/job/DocDev70/

### CODE REVIEW

File: docs/EN_US/HPCCSpark/SparkHPCC.xml [hunk 1]
```diff
@@ -438,6 +440,53 @@
...
+          </listitem>
+
+          <listitem>
+            <para><ulink
+            url="https://spark.apache.org/docs/latest/configuration.html#environment-variables ">https://spark.apache.org/docs/latest/configuration.html#environment-variables</ulink></para>
+          </listitem>
+        </itemizedlist>
+
+        <sect3 id="SampleSparkConfigurations">
+          <title>Sample Optional Configurations</title>
```
Comment (jpmcmu) [inline docs/EN_US/HPCCSpark/SparkHPCC.xml]: Would "Example Uses Cases" be a better title for this section>

File: docs/EN_US/HPCCSpark/SparkHPCC.xml [hunk 2]
```diff
@@ -438,6 +440,53 @@
       your Integrated Spark Master node's IP address.</para>
 
       <programlisting>https://192.168.56.101:8080</programlisting>
+
+      <sect2 id="Addl_Spark_Config">
+        <title>Integrated Spark Cluster Configuration Options</title>
```
Comment (jpmcmu) [inline docs/EN_US/HPCCSpark/SparkHPCC.xml]: I think we should probably mention that these configuration options are in addition to the configuration options available through config manager, and are meant for edge cases and more advanced setup.

File: docs/EN_US/HPCCSpark/SparkHPCC.xml [hunk 3]
```diff
@@ -438,6 +440,53 @@
...
+
+        <para>To customize your Integrated Spark cluster environment utilize
+        the provided <emphasis role="bold">spark-env.sh</emphasis> script.
+        Using the spark-env.sh script you can specify additional configuration
+        options for your Integrated Spark cluster.</para>
+
+        <programlisting>/etc/HPCCSystems/externals/spark/spark-env.sh </programlisting>
```
Comment (jpmcmu) [inline docs/EN_US/HPCCSpark/SparkHPCC.xml]: This script won't exist by default, and the user will have to create it. Should we mention this?

**Review by jpmcmu — CHANGES_REQUESTED** [review]
@g-pan Hey greg a few questions / comments

File: docs/EN_US/HPCCSpark/SparkHPCC.xml (hunk 1 above)
Comment (g-pan) [reply docs/EN_US/HPCCSpark/SparkHPCC.xml]: sure

File: docs/EN_US/HPCCSpark/SparkHPCC.xml (hunk 2 above)
Comment (g-pan) [reply docs/EN_US/HPCCSpark/SparkHPCC.xml]: good feedback. I will add to the doc.

File: docs/EN_US/HPCCSpark/SparkHPCC.xml (hunk 3 above)
Comment (g-pan) [reply docs/EN_US/HPCCSpark/SparkHPCC.xml]: yes. will do.

Comment (rpastrana) [reply docs/EN_US/HPCCSpark/SparkHPCC.xml]: I think @Michael-Gardner might have made a change which automatically adds this script (at least as of 7.0.12) Mike can you confirm?

Comment (jpmcmu) [reply docs/EN_US/HPCCSpark/SparkHPCC.xml]: @g-pan The above path changed from /etc/HPCCSystems/externals/spark/spark-env.sh to /etc/HPCCSystems/externals/spark-hadoop/spark-env.sh

**Comment by g-pan** [comment]
@rpastrana @jpmcmu could you please review the changes. Thanks.

**Comment by jpmcmu** [comment]
@g-pan @stuartort looks good

**Comment by rpastrana** [comment]
@g-pan reviewed offline pdf and approved

**Review by rpastrana — APPROVED** [review]

**Pull request merged**